ocs/volumes.py

```diff
@@ -0,0 +1,257 @@
+        name (str): name of the RBD volume
+        namepsace (str): namespace to create RBD volume
+
+    """
+    def __init__(self, name=None, namespace='default'):
+        KubeVolume.__init__(self, name, namespace)
+        self.kind = kinds.CEPHBLOCKPOOL
+        self.api_version = defaults.OPENSHIFT_REST_CLIENT_API_VERSION
+        self.service_cbp = self.dyn_client.resources.get(
+            api_version=self.api_version,
+            kind=self.kind
+        )
+        self.template_path = os.path.join(TEMPLATE_DIR, "cephblockpool.yaml")
+
+    def create_cephblockpool(
+        self,
+        cephblockpool_name_prefix="autotests-cephblockpool",
+        failureDomain="host",
+        size="3"
+    ):
+        """
+        Creates cephblock pool
+
+        Args:
+            cephblockpool_name_prefix (str): prefix given to cephblockpool
+            failureDomain (str): The failure domain across which the
+                                   replicas or chunks of data will be spread
+            size (int): The number of copies of the data in the pool.
+
+        Returns:
+            str : name of the cephblockpool created
+
+        Raises:
+            KeyError when error occured
+
+        Examples:
+            create_cephblockpool(
+                cephblockpool_name_prefix="autotests-blockpool",
+                failureDomain="host",
+                size=3)
+
+        """
+        if self.name:
+            cbp_name = self.name
+        else:
+            cbp_name = f"{cephblockpool_name_prefix}-{get_random_str()}"
+        # overwrite the namespace with openshift-storage, since cephblockpool
+        # is tied-up with openshift-storage
+        self.namespace = defaults.ROOK_CLUSTER_NAMESPACE
+
+        cephblockpool_data = {}
+        cephblockpool_data['cephblockpool_name'] = cbp_name
+        cephblockpool_data['rook_api_version'] = defaults.ROOK_API_VERSION
+        cephblockpool_data['failureDomain'] = failureDomain
+        cephblockpool_data['size'] = size
+
+        data = generate_yaml_from_jinja2_template_with_data(
+            self.template_path,
+            **cephblockpool_data
+        )
+        self.service_cbp.create(body=data, namespace=self.namespace)
+
+        return cbp_name
+
+
+class StorageClass(KubeVolume):
+    """
+    Class which contains StorageClass related functionality
+
+    Attributes:
+        name (str): name of the RBD volume
+        namepsace (str): namespace to create RBD volume
+
+    """
+    def __init__(self, name=None, namespace='default'):
+        KubeVolume.__init__(self, name, namespace)
+        self.kind = kinds.STORAGECLASS
+        self.api_version = defaults.OPENSHIFT_REST_CLIENT_API_VERSION
+        self.service_sc = self.dyn_client.resources.get(
+            api_version=self.api_version,
+            kind=self.kind
+        )
+        self.template_path = os.path.join(TEMPLATE_DIR, "storageclass.yaml")
+
+    def create_storageclass(
+        self,
+        blockPool,
```

Comment:
so this is a block storageClass?

Reply:
yes, that is block storage class.

Reply:
May be we can rename this class to BlockStorageClass or RBDStorageClass , not in this PR.

Reply:
+1 on renaming.
If we will have also cephfs storage class it will confuse ppl

Reply:
Intention of keeping the create_storageclass is in future I would like to use the same for creating cephfs also by passing different parameter( I didn't try creating cepfs manually. At this point of time not sure we can use the same for both or not ). But if we are not going to use the same function for both , then I will change the names accordingly like create_storageclass_rbd and create_storageclass_cephfs . I will address this in another PR